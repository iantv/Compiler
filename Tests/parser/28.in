void main(){
	int a = 0;
	if (a == 1){
		a += 'z';
	} else if (a == 0){
		a -= 'z';
	}
}