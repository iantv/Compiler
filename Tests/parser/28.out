main: function
	{
		a: variable
			int
		expression:
				0
			=
				a
		if: cond stmt
			exec cond:
					1
				==
					a
			{
				expression:
							z
						int
					+=
						a
			}
		else:
			{
				if: cond stmt
					exec cond:
							0
						==
							a
					{
						expression:
									z
								int
							-=
								a
					}
			}
	}
returns
	void
